src/button/Button.tsx

```diff
@@ -1,16 +1,40 @@
 import { Button as AriaButton } from "react-aria-components";
 
+import { Box } from "../../styled-system/jsx";
 import { button } from "../../styled-system/recipes/button";
+import { Spinner } from "../common/Spinner";
 
 import type { FC } from "react";
 import type { AriaButtonProps } from "react-aria";
 
 import type { ButtonVariantProps } from "../../styled-system/recipes";
 import type { WithoutClassName } from "../types";
 
-export type ButtonProps = WithoutClassName<AriaButtonProps> & Partial<ButtonVariantProps>;
+export type ButtonProps = WithoutClassName<AriaButtonProps> &
+  Partial<ButtonVariantProps> & {
+    /**
+     * If true, shows a loading spinner and disables the button.
+     */
+    isLoading?: boolean;
```

Comment:
Don't love this... we should be determining if in loading state if the promise in passed in onPressed has not resolved. This is very manual.